src/AutoRest.CSharp/Common/Output/Models/Serialization/Bicep/BicepObjectSerialization.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.CSharp.Common.Output.Models.Types;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Output.Builders;
+
+namespace AutoRest.CSharp.Output.Models.Serialization.Bicep
+{
+    internal class BicepObjectSerialization
+    {
+        public BicepObjectSerialization(string name, SerializableObjectType model)
+        {
+            Type = model.Type;
+            var enumeration = model.EnumerateHierarchy().SelectMany(o => o.Properties);
+            var properties = new List<BicepPropertySerialization>();
+            foreach (var property in enumeration)
+            {
+                BicepSerialization bicepSerialization;
+
+                if (property.SchemaProperty != null)
+                {
+                    bicepSerialization = SerializationBuilder.BuildBicepSerialization(
+                        property.SchemaProperty.Schema,
+                        property.Declaration.Type,
+                        false);
+                }
+                else if (property.InputModelProperty != null)
+                {
+                    bicepSerialization = SerializationBuilder.BuildBicepSerialization(property.InputModelProperty!.Type,
+                        property.Declaration.Type, false, property.SerializationFormat);
+                }
+                else if (property.Declaration.Name == "AdditionalProperties")
```

Comment:
Thanks, I reworked the implementation to reuse the JsonObjectSerialization types to avoid having to handle all of these complexities that have already been solved.